src/pudl/analysis/timeseries_cleaning.py

```diff
@@ -2047,35 +2098,47 @@ def _score_imputation(
 
         This takes the real output asset and the simulated output asset, and will
         compute a metric comparing the imputed simulated data to the real data. The
-        metric used is ``mean_absolute_percentage_error`` as percent error is more robust
-        to magnitude changes in the underlying data than total error.
+        metric used is ``mean_absolute_percentage_error`` as percent error is more
+        robust to magnitude changes in the underlying data than total error.
         """
         mape_dict = {}
         for group_name, gdf in simulated_df.groupby("simulation_group"):
             # Get just rows where we simultated NULLS
             simulated_gdf = gdf[gdf["flags"] == "simulated"]
 
             # Combine with real data
-            combined_df = simulated_gdf.merge(imputed_df, on=["datetime_utc", "id_col"])
+            combined_df = simulated_gdf.merge(
+                imputed_df,
+                how="inner",
+                on=["datetime_utc", "id_col"],
+                validate="one_to_one",
+            )
 
             # Compute metric
             mape_dict[group_name] = mean_absolute_percentage_error(
-                combined_df["value_col_x"],
```

Comment:
Was accidentally comparing reported values for a second which produces 0 MAPE, but fixed that now.